cpp/src/arrow/engine/substrait/options.h

```diff
@@ -65,16 +69,27 @@ using NamedTableProvider =
     std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
 static NamedTableProvider kDefaultNamedTableProvider;
 
+class ARROW_ENGINE_EXPORT ExtensionProvider {
+ public:
+  static std::shared_ptr<ExtensionProvider> kDefaultExtensionProvider;
```

Comment:
Should we define this the same way that `kDefaultNamedTableProvider` is defined? I don't see a reason why those two should be different

Reply:
Not sure, but I think it's fine to leave as is in this PR.